### DISCUSSION TITLE
ggwave-wasm.js

### DISCUSSION THREAD

**Discussion by twtom1990** [ggerganov/ggwave#7] [Q&A] [answered]
After I do cmake  and emcmake to this project. I open index.html to run the example , but there is no ggwave-wasm.js found, How can I build this javascript file ? I'm fully appreciated if someone could help me~~

**Comment by ggerganov** [comment, marked as answer]
You won't be able to simply open the `index.html` file. You have to serve it with a webserver. See the following link for more information: https://emscripten.org/docs/getting_started/Tutorial.html#generating-html

**Reply by twtom1990** [reply]
oh,thank you for the answer, but I still confused.
I just want to run it as the example ggwave-wasm . Should I create a directory named `build`  below `ggwave` , and then `cd build` , `emcmake cmake ..`. After that I’ll get a new directory `/ggwave` , there are some files below this directory such as `main.js `,`style.css`,`plucky.mp3`,`index.html`... Should I put these files into Apache and run this server?

I’m very appreciated if you could help me!🌸🌸

**Reply by ggerganov** [reply]
Yes, that's about right.

**Reply by twtom1990** [reply]
Thank you for the answer, I did it! I’m looking forward to implement the same function with java ☀️

**Reply by ggerganov** [reply]
Good luck! Feel free to contribute to this project if you feel like it
